### PR TITLE
fix: page header scroll and incidentals

### PR DESCRIPTION
**Pull request by lee-chase**
Contributes to #3347 

#### What did you change?

- Page header offset calculation
- Page header top value (fixes misaligned scroll of ActionButtons with a breadcrumb).
- BreadcrumbWithOverflow to use TooltipTrigger a button without styling that the Carbon Tooltip seems to now need. This fix is only visible on narrow screens and addresses the alignment of back arrow and title.
- UserProfileImage to also use TooltipTrigger
- Added TooltipTrigger

#### How did you test and verify your work?

Storybook.

### CODE REVIEW

**Review by elycheea — COMMENTED** [review]
Copyright dates on TooltipTrigger! 😤

But think you unintentionally left some changes in the PageHeader story and seeing a small issue in the tooltip padding of the user profile image. All fairly trivial changes so might be able to push these in a bit. 👀

File: packages/ibm-products/src/components/BreadcrumbWithOverflow/BreadcrumbWithOverflow.js [hunk 1]
```diff
@@ -306,15 +307,15 @@ export let BreadcrumbWithOverflow = ({
               <Link
                 href={backItem.href}
                 renderIcon={() => (
-                  <IconButton
-                    className={`${blockClass}__back__button`}
-                    align="right"
-                    kind="ghost"
+                  <Tooltip
+                    align="bottom"
```
[the remark this answers is not in the document]
Comment (elycheea) [reply packages/ibm-products/src/components/BreadcrumbWithOverflow/BreadcrumbWithOverflow.js:311]: Actually, seeing the difference in tooltip padding in this as well. Some changes locally that I could push but think probably better if we just do a quick check-in. 🙃

[the remark this answers is not in the document]
Comment (lee-chase) [reply packages/ibm-products/src/components/BreadcrumbWithOverflow/BreadcrumbWithOverflow.js]: Fixed this

File: packages/ibm-products/src/components/PageHeader/PageHeader.stories.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ import {
   TableCell,
   usePrefix,
   TabList,
+  Checkbox,
```
Comment (elycheea) [inline packages/ibm-products/src/components/PageHeader/PageHeader.stories.js:31]:
```suggestion
```

File: packages/ibm-products/src/components/PageHeader/PageHeader.stories.js [hunk 3]
```diff
@@ -816,6 +817,7 @@ const TemplateDemo = ({
 }) => {
   const carbonPrefix = usePrefix();
   const [isSideNavExpanded, setIsSideNavExpanded] = useState(false);
+  const [isChecked, setIsChecked] = useState(false);
```
Comment (elycheea) [inline packages/ibm-products/src/components/PageHeader/PageHeader.stories.js:820]:
```suggestion
```

**Review by elycheea — APPROVED** [review]

**Pull request merged**